src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/PersonHA.java

```diff
@@ -0,0 +1,42 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.message.EdifactValidationException;
+
+import java.util.Objects;
+
+@Builder
+@Data
+public class PersonHA extends Segment {
```

Comment:
This shouldn't be named `PersonHA` - you implemented NAD+FHS which is in message header and specifies 
> To identify the sender and recipient of the message.
> This segment is used to provide the code of the FHSA that is sending or receiving the message.

Reply:
changed that into 'PartyQualifier'